ui/StatusQ/sanity_checker/main.cpp

```diff
@@ -0,0 +1,45 @@
+#include <QCoreApplication>
+#include <QDebug>
+#include <QDirIterator>
+#include <QQmlComponent>
+#include <QQmlEngine>
+
+int main(int argc, char *argv[])
+{
+    QCoreApplication app(argc, argv);
+    QQmlEngine engine;
+
+    engine.addImportPath(QStringLiteral(":/"));
+
+    QDirIterator it(":", QDirIterator::Subdirectories);
+
+    while (it.hasNext()) {
+        QFileInfo info = it.fileInfo();
+
+        if (info.suffix() == QStringLiteral("qml")) {
+            QFile file(it.filePath());
+            file.open(QIODevice::ReadOnly);
+
+            QTextStream in(&file);
+            QString line = in.readLine();
+
+            if (line != QStringLiteral("pragma Singleton")) {
+                engine.setBaseUrl(QStringLiteral("qrc") + info.dir().path()
+                                  + QDir::separator());
+
+                QQmlComponent component(&engine, it.fileName());
+
+                if (component.isError()) {
+                    qDebug() << component.errors();
+                    return EXIT_FAILURE;
```

Comment:
Maybe instead of returning here immediately, we should allow all errors to be printed and then return, so that in case of many errors `sanity_checker` could be run only once.

Reply:
Good point. Fixed.